internal/kuberesource/sets.go

```diff
@@ -627,3 +627,159 @@ sleep inf
 
 	return []any{vss}
 }
+
+// MySQL returns the resources for deploying a MySQL database
+// with an encrypted luks volume using the workload-secret.
+func MySQL() []any {
+	initCommand := `#!/bin/bash
+# cryptsetup complains if there is no /run directory.
+mkdir /run
+set -e
+# device is the path to the block device to be encrypted.
+device="/dev/csi0"
+# workload_secret_path is the path to the Contrast workload secret.
+workload_secret_path="/contrast/secrets/workload-secret-seed"
+# tmp_key_path is the path to a temporary key file.
+tmp_key_path="/dev/shm/key"
+# disk_encryption_key_path is the path to the disk encryption key.
+disk_encryption_key_path="/dev/shm/disk-key"
+
+# If the device is not already a LUKS device, format it.
+if ! cryptsetup isLuks "${device}"; then
+	# Generate a random key for the first initialization.
+	dd if=/dev/urandom bs=1 count=32 2>/dev/null | base64 | tr -d '\n' > "${tmp_key_path}"
+	cryptsetup luksFormat --pbkdf-memory=10240 $device "${tmp_key_path}" </dev/null
+
+	# Now we can get the LUKS UUID and deterministically derive the disk encryption key from the workload secret.
+	uuid=$(blkid "${device}" -s UUID -o value)
+	openssl kdf -keylen 32 -kdfopt digest:SHA2-256 -kdfopt key:$(cat "${workload_secret_path}") \
```

Comment:
Can we use a key file instead of passing the key as arg? I just dumped it into logs accidentally by setting `-x`.

Reply:
`openssl kdf` only supports a key from the command line, but I don't think it's a big problem. I noticed though, that the option `hexkey` is probably more fitting here, since the workload secret is stored in hex.

Reply:
That's unfortunate, but further drives my point that we should highlight that this is demo code. 

(Not sure whether it was clear in my original comment: my concern is that the secrets end up in logs, which are readable by the infrastructure owner. _This is not the case in your code_, but a simple change like putting `-x` at the top is enough to let it happen, so I'd consider it a bit dangerous)